File: 1004/1004.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   1004.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2022/09/05 22:02:17 by jihoolee          #+#    #+#             */
/*   Updated: 2022/09/06 11:57:23 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <utility>

int square(int num) {
  return num * num;
}

bool in_circle(const std::pair<int, int>& from,
                const std::pair<int, int>& planet_center,
                int radius) {
  return (square(from.first - planet_center.first)
          + square(from.second - planet_center.second)) < square(radius);
}

int get_crossings(void) {
  std::pair<int, int> from;
  std::pair<int, int> dest;
  std::pair<int, int> planet_center;
  int radius;
  int planet_count;
  int result = 0;

  std::cin >> from.first >> from.second >> dest.first >> dest.second;
  std::cin >> planet_count;
  for (int i = 0; i < planet_count; ++i) {
    std::cin >> planet_center.first >> planet_center.second >> radius;
    if (in_circle(from, planet_center, radius)
        != in_circle(dest, planet_center, radius))
      ++result;
  }
  return result;
}

int main(void) {
  int cases;

  std::cin >> cases;
    for (int idx = 0; idx < cases; ++idx)
      std::cout << get_crossings() << std::endl;
  return 0;
}

File: 1004/1004.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    1004.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2022/09/06 11:58:03 by jihoolee          #+#    #+#              #
#    Updated: 2022/09/06 12:46:27 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

def square(num):
    return num * num

def in_circle(x1, y1, x2, y2, radius):
    return ((square(x1 - x2) + square(y1 - y2)) < square(radius))

def get_crossings():
    src_x, src_y, dest_x, dest_y = map(int, input().split())
    cases = int(input())
    result = 0
    for _ in range(cases):
        planet_center_x, planet_center_y, radius = map(int, input().split())
        if (in_circle(src_x, src_y, planet_center_x, planet_center_y, radius)
            != in_circle(dest_x, dest_y, planet_center_x, planet_center_y, radius)):
            result += 1
    return result

def main():
    inputs = int(input())
    for _ in range(inputs):
        print(get_crossings())

if (__name__ == '__main__'):
    main()
